File: jiheui/countAndSay.py
class Solution:
    def countAndSay(self, n: int) -> str:
        result=''
        while n>0:
            if not result:
                result = "1"
            else:
                elements=[]
                for i in result:
                    if not elements:
                        elements.append((i,1))
                    else:
                        key,value=elements[-1]
                        if key == i:
                            elements.pop(-1)
                            elements.append((i,value+1))
                        else:
                            elements.append((i,1))
                result=''
                for j in elements:
                    key,value=j
                    result+=str(value)+key
            n-=1
        return result


            







solution = Solution()
print(solution.countAndSay(8))

File: wonhyung/CountAndSay.php
<?php

namespace codingtest\wonhyung;

class CountAndSay
{
    function countAndSay1($n) {
        $string = "1";
        for ($i = 1; $i < $n; ++$i) {
            $tempString = "";
            $previousString = $string[0];
            $count = 1;
            for ($j = 1; $j < strlen($string); ++$j) {
                if ($string[$j] == $previousString) {
                    ++$count;
                } else {
                    $tempString .= $count . $previousString;
                    $count = 1;
                    $previousString = $string[$j];
                }
            }
            $string = $tempString . $count . $previousString;
        }

        return $string;
    }

    public function countAndSay2($n){
        if($n == 1){
            return "1";
        } else {
            $string = $this->countAndSay2($n-1);
            $result = "";
            $previousString = $string[0];
            $count = 0;
            for($i = 0;$i < strlen($string);$i++){
                if($string[$i] !== $previousString){
                    $result .= $count . $previousString;
                    $previousString = $string[$i];
                    $count = 1;
                } else {
                    $count++;
                }
            }
            $result .= $count . $previousString;
            return $result;
        }
    }
}